realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -378,8 +437,14 @@ void RealSenseNodeFactory::resetAndShutdown()
     ROS_INFO_STREAM(__FILE__ << " " << __LINE__ << " realsense_camera: device: " << _usb_port_id << " shutting down in 5s");
     sleep(5);
     ROS_INFO_STREAM(__FILE__ << " " << __LINE__ << " realsense_camera: device: " << _usb_port_id << " shutting down ROS and exiting.");
+    if (ros::isShuttingDown())
+    {
+        return; // someone else is shutting us down
+    }
+    ros::requestShutdown();
```

Comment:
If we plan to sleep on every request to shutdown we should wrap this in some utility method.